posthog/test/test_migration_0216.py

```diff
@@ -0,0 +1,36 @@
+from django.db import IntegrityError
+
+from posthog.test.base import NonAtomicTestMigrations
+
+
+class TaggedItemsUniquenessTest(NonAtomicTestMigrations):
+
+    migrate_from = "0215_add_tags_back"  # type: ignore
+    migrate_to = "0216_uniqueness_constraint_tagged_items"  # type: ignore
+
+    def setUpBeforeMigration(self, apps):
+        Dashboard = apps.get_model("posthog", "Dashboard")
+        Tag = apps.get_model("posthog", "Tag")
+        TaggedItem = apps.get_model("posthog", "TaggedItem")
+
+        # Setup
+        self.tag = Tag.objects.create(name="tag", team_id=self.team.id)
+        self.dashboard = Dashboard.objects.create(team_id=self.team.id, name="private dashboard")
+
+        # Before migration you can create duplicate tagged items
+        taggeditem_1 = TaggedItem.objects.create(dashboard_id=self.dashboard.id, tag_id=self.tag.id)
+        taggeditem_2 = TaggedItem.objects.create(dashboard_id=self.dashboard.id, tag_id=self.tag.id)
+        self.assertNotEqual(taggeditem_1.id, taggeditem_2.id)
+
+        # Remove duplicate row so that unique index can be created
+        taggeditem_2.delete()
+
+    def test_taggeditems_uniqueness(self):
+        TaggedItem = self.apps.get_model("posthog", "TaggedItem")  # type: ignore
+
+        with self.assertRaises(IntegrityError):
+            TaggedItem.objects.create(dashboard_id=self.dashboard.id, tag_id=self.tag.id)
```

Comment:
Given it's very low effort I would add in tests for each of the model types.

non-blocking: not for this PR, but is there a more stable API level we can test at? e.g. at the REST api level, I would add something like:

```
def test_tagging_items_are_unique():
    PUT /api/insights/abc/tags/some-tag
    PUT /api/insights/abc/tags/some-tag
    GET /api/insights/abc/tags/

    assert there is just one tag
```

It's worth noting that this isn't just a test related to this specific migration, but a requirement/feature of the API

Reply:
Good point, that would've probably been an easier test to write than the migration test. I'll add one test for each object.

Reply:
Also probably a good idea to add model level tests too.